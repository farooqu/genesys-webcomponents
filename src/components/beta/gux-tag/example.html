<h1>gux-tag-beta</h1>
<h2>Tags</h2>
<div class="tags">
  <gux-tag-beta>default</gux-tag-beta>
  <gux-tag-beta color="default">default (explicit)</gux-tag-beta>
  <gux-tag-beta color="navy">navy</gux-tag-beta>
  <gux-tag-beta color="blue">blue</gux-tag-beta>
  <gux-tag-beta color="electric-purple">electric-purple</gux-tag-beta>
  <gux-tag-beta color="aqua-green">aqua-green</gux-tag-beta>
  <gux-tag-beta color="fuscha">fuscha</gux-tag-beta>
  <gux-tag-beta color="bubblegum-pink">bubblegum-pink</gux-tag-beta>
  <gux-tag-beta color="dark-purple">dark purple</gux-tag-beta>
  <gux-tag-beta color="olive-green">olive-green</gux-tag-beta>
  <gux-tag-beta color="lilac">lilac</gux-tag-beta>
  <gux-tag-beta color="alert-yellow-green">alert-yellow-green</gux-tag-beta>
  <gux-tag-beta color="invalid">invalid</gux-tag-beta>
</div>

<h2>Disabled tags</h2>
<div class="tags">
  <gux-tag-beta disabled>default</gux-tag-beta>
  <gux-tag-beta color="default" disabled>default (explicit)</gux-tag-beta>
  <gux-tag-beta color="navy" disabled>navy</gux-tag-beta>
  <gux-tag-beta color="blue" disabled>blue</gux-tag-beta>
  <gux-tag-beta color="electric-purple" disabled>electric-purple</gux-tag-beta>
  <gux-tag-beta color="aqua-green" disabled>aqua-green</gux-tag-beta>
  <gux-tag-beta color="fuscha" disabled>fuscha</gux-tag-beta>
  <gux-tag-beta color="bubblegum-pink" disabled>bubblegum-pink</gux-tag-beta>
  <gux-tag-beta color="dark-purple" disabled>dark purple</gux-tag-beta>
  <gux-tag-beta color="olive-green" disabled>olive-green</gux-tag-beta>
  <gux-tag-beta color="lilac" disabled>lilac</gux-tag-beta>
  <gux-tag-beta color="alert-yellow-green" disabled
    >alert-yellow-green</gux-tag-beta
  >
  <gux-tag-beta color="invalid" disabled>invalid</gux-tag-beta>
</div>

<h2>Tag icons</h2>
<div class="tags">
  <gux-tag-beta icon="bolt">default</gux-tag-beta>
  <gux-tag-beta icon="bolt" color="default">default (explicit)</gux-tag-beta>
  <gux-tag-beta icon="bolt" color="navy">navy</gux-tag-beta>
  <gux-tag-beta icon="bolt" color="blue">blue</gux-tag-beta>
  <gux-tag-beta icon="bolt" color="electric-purple"
    >electric-purple</gux-tag-beta
  >
  <gux-tag-beta icon="bolt" color="aqua-green">aqua-green</gux-tag-beta>
  <gux-tag-beta icon="bolt" color="fuscha">fuscha</gux-tag-beta>
  <gux-tag-beta icon="bolt" color="bubblegum-pink" disabled
    >bubblegum-pink</gux-tag-beta
  >
  <gux-tag-beta icon="bolt" color="dark-purple" disabled
    >dark purple</gux-tag-beta
  >
  <gux-tag-beta icon="bolt" color="olive-green" disabled
    >olive-green</gux-tag-beta
  >
  <gux-tag-beta icon="bolt" color="lilac" disabled>lilac</gux-tag-beta>
  <gux-tag-beta icon="bolt" color="alert-yellow-green" disabled
    >alert-yellow-green</gux-tag-beta
  >
  <gux-tag-beta icon="bolt" color="invalid" disabled>invalid</gux-tag-beta>
</div>

<h2>Removable tags</h2>
<div class="remove-tag" class="tags">
  <gux-tag-beta value="1" removable>default</gux-tag-beta>
  <gux-tag-beta color="default" value="2" removable
    >default (explicit)</gux-tag-beta
  >
  <gux-tag-beta color="navy" value="3" removable>navy</gux-tag-beta>
  <gux-tag-beta color="blue" value="4" removable>blue</gux-tag-beta>
  <gux-tag-beta color="electric-purple" value="5" removable
    >electric-purple</gux-tag-beta
  >
  <gux-tag-beta color="aqua-green" value="6" removable>aqua-green</gux-tag-beta>
  <gux-tag-beta color="fuscha" value="7" removable>fuscha</gux-tag-beta>
  <gux-tag-beta color="bubblegum-pink" value="8" removable disabled
    >bubblegum-pink</gux-tag-beta
  >
  <gux-tag-beta color="dark-purple" value="9" removable disabled
    >dark purple</gux-tag-beta
  >
  <gux-tag-beta color="olive-green" value="10" removable disabled
    >olive-green</gux-tag-beta
  >
  <gux-tag-beta color="lilac" value="11" removable disabled>lilac</gux-tag-beta>
  <gux-tag-beta color="alert-yellow-green" value="12" removable disabled
    >alert-yellow-green</gux-tag-beta
  >
  <gux-tag-beta color="invalid" value="13" removable disabled
    >invalid</gux-tag-beta
  >
</div>

<h2>Icons and Removable tags</h2>
<div class="remove-tag" class="tags">
  <gux-tag-beta icon="bolt" value="1" removable>default</gux-tag-beta>
  <gux-tag-beta icon="bolt" color="default" value="2" removable
    >default (explicit)</gux-tag-beta
  >
  <gux-tag-beta icon="bolt" color="navy" value="3" removable>navy</gux-tag-beta>
  <gux-tag-beta icon="bolt" color="blue" value="4" removable>blue</gux-tag-beta>
  <gux-tag-beta icon="bolt" color="electric-purple" value="5" removable
    >electric-purple</gux-tag-beta
  >
  <gux-tag-beta icon="bolt" color="aqua-green" value="6" removable
    >aqua-green</gux-tag-beta
  >
  <gux-tag-beta icon="bolt" color="fuscha" value="7" removable
    >fuscha</gux-tag-beta
  >
  <gux-tag-beta icon="bolt" color="bubblegum-pink" value="8" removable disabled
    >bubblegum-pink</gux-tag-beta
  >
  <gux-tag-beta icon="bolt" color="dark-purple" value="9" removable disabled
    >dark purple</gux-tag-beta
  >
  <gux-tag-beta icon="bolt" color="olive-green" value="10" removable disabled
    >olive-green</gux-tag-beta
  >
  <gux-tag-beta icon="bolt" color="lilac" value="11" removable disabled
    >lilac</gux-tag-beta
  >
  <gux-tag-beta
    icon="bolt"
    color="alert-yellow-green"
    value="12"
    removable
    disabled
    >alert-yellow-green</gux-tag-beta
  >
  <gux-tag-beta icon="bolt" color="invalid" value="13" removable disabled
    >invalid</gux-tag-beta
  >
</div>

<h2>Icons and Removable tags localised</h2>
<div class="remove-tag" class="tags" lang="de">
  <gux-tag-beta icon="bolt" value="1" removable>default</gux-tag-beta>
  <gux-tag-beta icon="bolt" color="default" value="2" removable
    >default (explicit)</gux-tag-beta
  >
  <gux-tag-beta icon="bolt" color="navy" value="3" removable>navy</gux-tag-beta>
  <gux-tag-beta icon="bolt" color="blue" value="4" removable>blue</gux-tag-beta>
  <gux-tag-beta icon="bolt" color="electric-purple" value="5" removable
    >electric-purple</gux-tag-beta
  >
  <gux-tag-beta icon="bolt" color="aqua-green" value="6" removable
    >aqua-green</gux-tag-beta
  >
  <gux-tag-beta icon="bolt" color="fuscha" value="7" removable
    >fuscha</gux-tag-beta
  >
  <gux-tag-beta icon="bolt" color="bubblegum-pink" value="8" removable disabled
    >bubblegum-pink</gux-tag-beta
  >
  <gux-tag-beta icon="bolt" color="dark-purple" value="9" removable disabled
    >dark purple</gux-tag-beta
  >
  <gux-tag-beta icon="bolt" color="olive-green" value="10" removable disabled
    >olive-green</gux-tag-beta
  >
  <gux-tag-beta icon="bolt" color="lilac" value="11" removable disabled
    >lilac</gux-tag-beta
  >
  <gux-tag-beta
    icon="bolt"
    color="alert-yellow-green"
    value="12"
    removable
    disabled
    >alert-yellow-green</gux-tag-beta
  >
  <gux-tag-beta icon="bolt" color="invalid" value="13" removable disabled
    >invalid</gux-tag-beta
  >
</div>

<style
  onload="(function() {
  const removeTagElements = document.getElementsByClassName('remove-tag');

  Array.from(removeTagElements).forEach(function(removeTagElement) {
    removeTagElement.addEventListener('guxdelete', (event) => {
      const valueToRemove = event.detail;

      Array.from(
        removeTagElement.querySelectorAll('gux-tag-beta')
      ).forEach((tag) => {
        if (tag.value === valueToRemove) {
          tag.remove();
        }
      });
    })
  });
})()"
>
  .tags {
    display: flex;
    flex-wrap: wrap;
    padding: 10px;
    border: 1px solid var(--gux-charcoal-grey);
  }

  gux-tag-beta {
    margin: 4px;
  }
</style>
