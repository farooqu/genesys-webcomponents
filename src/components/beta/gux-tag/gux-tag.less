@import (reference) '../../../style/color.less';
@import (reference) '../../../style/mixins.less';
@import (reference) '../../../style/spacing.less';
@import (reference) '../../../style/typography.less';

// Style
:host {
  display: inline-block;
}

.gux-tag {
  display: flex;
  flex-direction: row;
  flex-wrap: nowrap;
  align-content: stretch;
  align-items: center;
  justify-content: flex-start;
  color: @gux-grey-100;
  background-color: @gux-black-50;
  border-radius: @spacing-2xs;

  .gux-tag-icon-container {
    display: flex;
    flex: 0 1 auto;
    align-content: center;
    align-items: center;
    align-self: auto;
    justify-content: center;
    width: 24px;
    height: 24px;
    margin-right: (-1 * @spacing-xs);

    .gux-tag-icon {
      width: 16px;
      height: 16px;
    }
  }

  .gux-tag-text {
    flex: 0 1 auto;
    align-self: auto;
    padding: 0 @spacing-xs;
    line-height: 24px;
    white-space: nowrap;

    .gux-sr-only {
      .gux-sr-only-clip();
    }
  }

  .gux-tag-remove-button {
    all: unset;
    display: flex;
    align-content: center;
    align-items: center;
    justify-content: center;
    width: 24px;
    height: 24px;
    margin-left: (-1 * @spacing-xs);
    overflow: hidden;

    &:not(:disabled):hover {
      cursor: pointer;
    }

    .gux-tag-remove-icon {
      width: 12px;
      height: 12px;
      border-radius: 25%;
    }

    &:focus-within {
      .gux-tag-remove-icon {
        .focus-ring-small();
      }
    }
  }

  &.gux-disabled {
    opacity: 0.5;
  }

  &.gux-default {
    background-color: @gux-black-50;
  }

  &.gux-navy {
    background-color: @gux-navy-sec;
  }

  &.gux-blue {
    color: @gux-black-50;
    background-color: @gux-blue-sec;
  }

  &.gux-electric-purple {
    background-color: @gux-electric-purple-sec;
  }

  &.gux-aqua-green {
    color: @gux-black-40;
    background-color: @gux-aqua-green-sec;
  }

  &.gux-fuscha {
    color: @gux-black-10;
    background-color: @gux-fuscha-sec;
  }

  &.gux-dark-purple {
    background-color: @gux-dark-purple-sec;
  }

  &.gux-bubblegum-pink {
    color: @gux-black-50;
    background-color: @gux-bubblegum-pink-sec;
  }

  &.gux-olive-green {
    color: @gux-black-20;
    background-color: @gux-olive-green-sec;
  }

  &.gux-lilac {
    color: @gux-black-50;
    background-color: @gux-lilac-sec;
  }

  &.gux-alert-yellow-green {
    color: @gux-black-50;
    background-color: @gux-alert-yellow-green-sec;
  }
}
