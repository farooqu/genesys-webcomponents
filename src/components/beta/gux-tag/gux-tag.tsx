import {
  Component,
  Element,
  Event,
  EventEmitter,
  h,
  JSX,
  Listen,
  Prop,
  State
} from '@stencil/core';

import { buildI18nForComponent, GetI18nValue } from '../../../i18n';
import { trackComponent } from '../../../usage-tracking';

import tagResources from './i18n/en.json';
import { GuxTagColor } from './gux-tag.types';

@Component({
  styleUrl: 'gux-tag.less',
  tag: 'gux-tag-beta',
  shadow: true
})
export class GuxTag {
  private i18n: GetI18nValue;

  @Element()
  root: HTMLElement;

  /**
   * Triggered when click on remove button
   */
  @Event()
  guxdelete: EventEmitter<string>;

  /**
   * Tag background color.
   */
  @Prop()
  color: GuxTagColor = 'default';

  /**
   * Index for remove tag
   */
  @Prop()
  value: string;

  /**
   * Tag is removable.
   */
  @Prop()
  disabled: boolean = false;

  /**
   * Tag is removable.
   */
  @Prop()
  removable: boolean = false;

  /**
   * Tag icon name.
   */
  @Prop()
  icon: string;

  @State()
  label: string;

  @Listen('keydown')
  onKeyDown(event: KeyboardEvent): void {
    switch (event.key) {
      case 'Backspace':
      case 'Delete':
        this.removeTag();
    }
  }

  private removeTag(): void {
    if (this.disabled || !this.removable) {
      return;
    }

    this.guxdelete.emit(this.value);
  }

  private onSlotChange(event: Event) {
    this.label = (
      event.composedPath()[0] as HTMLSlotElement
    ).assignedNodes()[0].textContent;
  }

  private renderIcon(): JSX.Element {
    if (this.icon) {
      return (
        <div class="gux-tag-icon-container">
          <gux-icon class="gux-tag-icon" icon-name={this.icon} decorative />
        </div>
      ) as JSX.Element;
    }
  }

  private renderText(): JSX.Element {
    return (
      <div class="gux-tag-text">
        <slot aria-hidden="true" onSlotchange={this.onSlotChange.bind(this)} />

        {!this.disabled && (
          <div class="gux-sr-only">
            {this.i18n('tag', { label: this.label })}
          </div>
        )}
        {this.disabled && (
          <div class="gux-sr-only">
            {this.i18n('tag-disabled', { label: this.label })}
          </div>
        )}
      </div>
    ) as JSX.Element;
  }

  private renderRemoveButton(): JSX.Element {
    if (this.removable) {
      return (
        <button
          class="gux-tag-remove-button"
          onClick={this.removeTag.bind(this)}
          type="button"
          disabled={this.disabled}
        >
          <gux-icon
            class="gux-tag-remove-icon"
            icon-name="close"
            screenreader-text={this.i18n('remove-tag', {
              label: this.label
            })}
          />
        </button>
      ) as JSX.Element;
    }
  }

  componentWillLoad(): void {
    trackComponent(this.root, {
      variant: this.removable ? 'removable' : 'permenant'
    });
  }

  async componentWillRender(): Promise<void> {
    this.i18n = await buildI18nForComponent(this.root, tagResources);
  }

  render(): JSX.Element {
    return (
      <div
        class={{
          'gux-tag': true,
          [`gux-${this.color}`]: true,
          'gux-disabled': this.disabled
        }}
        aria-disabled={this.disabled.toString()}
      >
        {this.renderIcon()}
        {this.renderText()}
        {this.renderRemoveButton()}
      </div>
    ) as JSX.Element;
  }
}
