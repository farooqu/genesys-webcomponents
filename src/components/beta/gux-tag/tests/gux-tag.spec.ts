import { newSpecPage } from '@stencil/core/testing';
import { GuxTag } from '../gux-tag';

const components = [GuxTag];
const language = 'en';

describe('gux-tag-beta', () => {
  describe('#render', () => {
    [
      '<gux-tag-beta>default</gux-tag-beta>',
      '<gux-tag-beta color="default">default (explicit)</gux-tag-beta>',
      '<gux-tag-beta color="navy">navy</gux-tag-beta>',
      '<gux-tag-beta icon="bolt" color="navy">navy</gux-tag-beta>',
      '<gux-tag-beta icon="bolt" color="navy" value="3" removable>navy</gux-tag-beta>',
      '<gux-tag-beta icon="bolt" color="navy" value="3" removable disabled>navy</gux-tag-beta>'
    ].forEach((html, index) => {
      it(`should render component as expected (${index + 1})`, async () => {
        const page = await newSpecPage({ components, html, language });

        expect(page.root).toMatchSnapshot();
      });
    });
  });

  describe('delete', () => {
    describe('click', () => {
      it('should not have a delete button if tag is not removable', async () => {
        const html = `<gux-tag-beta icon="bolt" color="navy" value="3">navy</gux-tag-beta>`;
        const page = await newSpecPage({ components, html, language });
        const element = page.root as HTMLElement;
        const deleteButton: HTMLButtonElement =
          element.shadowRoot.querySelector('.gux-tag-remove-button');

        expect(deleteButton).toBeNull();
      });

      it('should emit guxdelete if tag is removable and not disabled', async () => {
        const html = `<gux-tag-beta icon="bolt" color="navy" value="3" removable>navy</gux-tag-beta>`;
        const page = await newSpecPage({ components, html, language });
        const element = page.root as HTMLElement;
        const deleteButton: HTMLButtonElement =
          element.shadowRoot.querySelector('.gux-tag-remove-button');
        const guxdeleteSpy = jest.fn();

        page.win.addEventListener('guxdelete', guxdeleteSpy);

        deleteButton.click();
        await page.waitForChanges();

        expect(guxdeleteSpy).toHaveBeenCalledWith(
          expect.objectContaining({ detail: '3' })
        );
      });

      it('should not emit guxdelete if tag is removable and disabled', async () => {
        const html = `<gux-tag-beta icon="bolt" color="navy" value="3" removable disabled>navy</gux-tag-beta>`;
        const page = await newSpecPage({ components, html, language });
        const element = page.root as HTMLElement;
        const deleteButton: HTMLButtonElement =
          element.shadowRoot.querySelector('.gux-tag-remove-button');
        const guxdeleteSpy = jest.fn();

        page.win.addEventListener('guxdelete', guxdeleteSpy);

        deleteButton.click();
        await page.waitForChanges();

        expect(guxdeleteSpy).not.toHaveBeenCalled();
      });
    });

    describe('keypress', () => {
      it('should emit guxdelete if tag is focused and removable and "Delete" is pressed', async () => {
        const html = `<gux-tag-beta icon="bolt" color="navy" value="3" removable>navy</gux-tag-beta>`;
        const page = await newSpecPage({ components, html, language });
        const element = page.root as HTMLElement;
        const guxdeleteSpy = jest.fn();

        page.win.addEventListener('guxdelete', guxdeleteSpy);

        element.dispatchEvent(new KeyboardEvent('keydown', { key: 'Delete' }));
        await page.waitForChanges();

        expect(guxdeleteSpy).toHaveBeenCalledWith(
          expect.objectContaining({ detail: '3' })
        );
      });

      it('should emit guxdelete if tag is focused and removable and "Backspace" is pressed', async () => {
        const html = `<gux-tag-beta icon="bolt" color="navy" value="3" removable>navy</gux-tag-beta>`;
        const page = await newSpecPage({ components, html, language });
        const element = page.root as HTMLElement;
        const guxdeleteSpy = jest.fn();

        page.win.addEventListener('guxdelete', guxdeleteSpy);

        element.dispatchEvent(
          new KeyboardEvent('keydown', { key: 'Backspace' })
        );
        await page.waitForChanges();

        expect(guxdeleteSpy).toHaveBeenCalledWith(
          expect.objectContaining({ detail: '3' })
        );
      });

      it('should not emit guxdelete if tag is focused and not removable and "Delete" is pressed', async () => {
        const html = `<gux-tag-beta icon="bolt" color="navy" value="3">navy</gux-tag-beta>`;
        const page = await newSpecPage({ components, html, language });
        const element = page.root as HTMLElement;
        const guxdeleteSpy = jest.fn();

        page.win.addEventListener('guxdelete', guxdeleteSpy);

        element.dispatchEvent(new KeyboardEvent('keydown', { key: 'Delete' }));
        await page.waitForChanges();

        expect(guxdeleteSpy).not.toHaveBeenCalled();
      });

      it('should not emit guxdelete if tag is focused and not removable and "Backspace" is pressed', async () => {
        const html = `<gux-tag-beta icon="bolt" color="navy" value="3">navy</gux-tag-beta>`;
        const page = await newSpecPage({ components, html, language });
        const element = page.root as HTMLElement;
        const guxdeleteSpy = jest.fn();

        page.win.addEventListener('guxdelete', guxdeleteSpy);

        element.dispatchEvent(
          new KeyboardEvent('keydown', { key: 'Backspace' })
        );
        await page.waitForChanges();

        expect(guxdeleteSpy).not.toHaveBeenCalled();
      });
    });
  });
});
