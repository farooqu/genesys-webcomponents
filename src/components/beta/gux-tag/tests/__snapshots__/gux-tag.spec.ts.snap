// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`gux-tag-beta #render should render component as expected (1) 1`] = `
<gux-tag-beta>
  <mock:shadow-root>
    <div aria-disabled="false" class="gux-default gux-tag">
      <div class="gux-tag-text">
        <slot aria-hidden="true"></slot>
        <div class="gux-sr-only">
          tag with label:
        </div>
      </div>
    </div>
  </mock:shadow-root>
  default
</gux-tag-beta>
`;

exports[`gux-tag-beta #render should render component as expected (2) 1`] = `
<gux-tag-beta color="default">
  <mock:shadow-root>
    <div aria-disabled="false" class="gux-default gux-tag">
      <div class="gux-tag-text">
        <slot aria-hidden="true"></slot>
        <div class="gux-sr-only">
          tag with label:
        </div>
      </div>
    </div>
  </mock:shadow-root>
  default (explicit)
</gux-tag-beta>
`;

exports[`gux-tag-beta #render should render component as expected (3) 1`] = `
<gux-tag-beta color="navy">
  <mock:shadow-root>
    <div aria-disabled="false" class="gux-navy gux-tag">
      <div class="gux-tag-text">
        <slot aria-hidden="true"></slot>
        <div class="gux-sr-only">
          tag with label:
        </div>
      </div>
    </div>
  </mock:shadow-root>
  navy
</gux-tag-beta>
`;

exports[`gux-tag-beta #render should render component as expected (4) 1`] = `
<gux-tag-beta color="navy" icon="bolt">
  <mock:shadow-root>
    <div aria-disabled="false" class="gux-navy gux-tag">
      <div class="gux-tag-icon-container">
        <gux-icon class="gux-tag-icon" decorative="" icon-name="bolt"></gux-icon>
      </div>
      <div class="gux-tag-text">
        <slot aria-hidden="true"></slot>
        <div class="gux-sr-only">
          tag with label:
        </div>
      </div>
    </div>
  </mock:shadow-root>
  navy
</gux-tag-beta>
`;

exports[`gux-tag-beta #render should render component as expected (5) 1`] = `
<gux-tag-beta color="navy" icon="bolt" removable="" value="3">
  <mock:shadow-root>
    <div aria-disabled="false" class="gux-navy gux-tag">
      <div class="gux-tag-icon-container">
        <gux-icon class="gux-tag-icon" decorative="" icon-name="bolt"></gux-icon>
      </div>
      <div class="gux-tag-text">
        <slot aria-hidden="true"></slot>
        <div class="gux-sr-only">
          tag with label:
        </div>
      </div>
      <button class="gux-tag-remove-button" type="button">
        <gux-icon class="gux-tag-remove-icon" icon-name="close" screenreader-text="Remove tag with label: "></gux-icon>
      </button>
    </div>
  </mock:shadow-root>
  navy
</gux-tag-beta>
`;

exports[`gux-tag-beta #render should render component as expected (6) 1`] = `
<gux-tag-beta color="navy" disabled="" icon="bolt" removable="" value="3">
  <mock:shadow-root>
    <div aria-disabled="true" class="gux-disabled gux-navy gux-tag">
      <div class="gux-tag-icon-container">
        <gux-icon class="gux-tag-icon" decorative="" icon-name="bolt"></gux-icon>
      </div>
      <div class="gux-tag-text">
        <slot aria-hidden="true"></slot>
        <div class="gux-sr-only">
          disabled tag with label:
        </div>
      </div>
      <button class="gux-tag-remove-button" disabled="" type="button">
        <gux-icon class="gux-tag-remove-icon" icon-name="close" screenreader-text="Remove tag with label: "></gux-icon>
      </button>
    </div>
  </mock:shadow-root>
  navy
</gux-tag-beta>
`;
